golang/connection_test.go

```diff
@@ -168,5 +168,8 @@ func withTestChannel(t *testing.T, f func(ch *Channel)) {
 
 	go ch.ListenAndHandle()
 
-	f(ch)
+	func() {
```

Comment:
You don't need the wrapping func for the defer - you can just defer and then call f(ch).  The defer won't kick in until you exit the withTestChannel scope, which won't happen until f(ch) is complete.